### PR TITLE
udev: make udevadm test and test-builtin not destructive

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/udev/meson.build [hunk 1]
```diff
@@ -205,7 +205,9 @@ executables += [
         },
         udev_test_template + {
                 'sources' : files('net/test-link-config-tables.c'),
+                'include_directories' : includes + include_directories('.'),
                 'suite' : 'udev',
+
```
Comment (bluca) [inline src/udev/meson.build]: spurious newline?

**Review by bluca — APPROVED** [review]

**Comment by ddiss** [comment]
FWIW, we used the "destructive" `udevadm test` behavior to manually initialize `/sys/class/net/` devices prior to `systemd-network` invocation in lightweight (initramfs only) VMs.
It was working very well over the years, but fails as of 089bef66316e5bdc91b9984148e5a6455449c1da (bisected).

**Comment by ddiss** [comment]
> udevadm test is for testing. You should use udevadm trigger for processing specific device(s).

`udevadm test` could be used standalone, while `udevadm trigger` (AFAICT) requires a running udevd. I realize it perhaps wasn't a standard environment, but it was nice for lightweight VM images.

> Note, trigger command has --prioritized-subsystem=. So if you want to initialize network devices earlier, reorder prioritized subsystems in systemd-udev-trigger.service.

Thanks for the advice (and quick response)!

**Comment by yuwata** [comment]
> > udevadm test is for testing. You should use udevadm trigger for processing specific device(s).
> 
> `udevadm test` could be used standalone, while `udevadm trigger` (AFAICT) requires a running udevd. I realize it perhaps wasn't a standard environment, but it was nice for lightweight VM images.

Interesting. But, (at least currently) such use case is not supported.

**Pull request merged**